pulser/simulation/simresults.py

```diff
@@ -54,6 +59,194 @@ def __init__(self, run_output, dim, size, basis_name, meas_basis=None):
                     )
         self._meas_basis = meas_basis
 
+    @abstractmethod
+    def expect(self, obs_list):
+        """Calculates the expectation value of a list of observables.
+
+        Args:
+            obs_list (array-like of qutip.Qobj or array-like of numpy.ndarray):
+                A list of observables whose expectation value will be
+                calculated. If necessary, each member will be transformed into
+                a qutip.Qobj instance.
+        """
+        pass
+
+    @abstractmethod
+    def sample_state(self, t=-1, meas_basis=None, N_samples=1000):
+        r"""Returns the result of multiple measurements in a given basis.
+
+        The enconding of the results depends on the meaurement basis. Namely:
+
+        - *ground-rydberg* : :math:`1 = |r\rangle;~ 0 = |g\rangle, |h\rangle`
+        - *digital* : :math:`1 = |h\rangle;~ 0 = |g\rangle, |r\rangle`
+
+        Note:
+            The results are presented using a big-endian representation,
+            according to the pre-established qubit ordering in the register.
+            This means that, when sampling a register with qubits ('q0','q1',
+            ...), in this order, the corresponding value, in binary, will be
+            0Bb0b1..., where b0 is the outcome of measuring 'q0', 'b1' of
+            measuring 'q1' and so on.
+
+        Keyword Args:
+            meas_basis (str, default=None): 'ground-rydberg' or 'digital'. If
+                left as None, uses the measurement basis defined in the
+                original sequence.
+            N_samples (int, default=1000): Number of samples to take.
+
+        Raises:
+            ValueError: If trying to sample without a defined 'meas_basis' in
+                the arguments when the original sequence is not measured.
+        """
+        pass
+
+    @abstractmethod
+    def sample_final_state(self, meas_basis=None, N_samples=1000):
+        pass
+
+
+class NoisyResults(SimulationResults):
+    """
+    Results of a noisy simulation run of a pulse sequence. Contrary to a
+    CleanResults object, this object contains a unique Counter describing the
+    state distribution at the time it was created by using
+    sim.run(spam=True, t).
```

Comment:
Some comment as before, just use the first phrase on the first line a move the rest to the description.